src/zarr/core/metadata/v3.py

```diff
@@ -297,71 +332,99 @@ def update_attributes(self, attributes: dict[str, JSON]) -> Self:
         return replace(self, attributes=attributes)
 
 
+# enum Literals can't be used in typing, so we have to restate all of the V3 dtypes as types
+# https://github.com/python/typing/issues/781
+
 BOOL = np.bool_
-BOOL_DTYPE = np.dtypes.BoolDType
-INTEGER_DTYPE = (
-    np.dtypes.Int8DType
-    | np.dtypes.Int16DType
-    | np.dtypes.Int32DType
-    | np.dtypes.Int64DType
-    | np.dtypes.UInt8DType
-    | np.dtypes.UInt16DType
-    | np.dtypes.UInt32DType
-    | np.dtypes.UInt64DType
-)
+# BOOL_DTYPE = np.dtypes.BoolDType
+BOOL_DTYPE = Literal["bool"]
+INTEGER_DTYPE = Literal["int8", "int16", "int32", "int64", "uint8", "uint16", "uint32", "uint64"]
+# INTEGER_DTYPE = (
+#     np.dtypes.Int8DType
+#     | np.dtypes.Int16DType
+#     | np.dtypes.Int32DType
+#     | np.dtypes.Int64DType
+#     | np.dtypes.UInt8DType
+#     | np.dtypes.UInt16DType
+#     | np.dtypes.UInt32DType
+#     | np.dtypes.UInt64DType
+# )
```

Comment:
I am attempting to replace these numpy dtypes with our own internal dtypes using literals (trying to follow this advice - https://github.com/python/typing/issues/781).

It does not seem to be working.